src/gui/toolbarMenubar/ToolMenuHandler.cpp

```diff
@@ -372,10 +374,10 @@ void ToolMenuHandler::initToolItems()
 
 	//Icon snapping.svg made by www.freepik.com from www.flaticon.com
 	addToolItem(new ToolButton(listener, gui, "ROTATION_SNAPPING", ACTION_ROTATION_SNAPPING, GROUP_SNAPPING, false,
-								"snapping.svg", _("Rotation Snapping"), gui->get("menuSnapRotation")));
+								"snapping" + darkThemeExt + ".svg", _("Rotation Snapping"), gui->get("menuSnapRotation")));
 
 	addToolItem(new ToolButton(listener, gui, "GRID_SNAPPING", ACTION_GRID_SNAPPING, GROUP_GRID_SNAPPING, false,
-								"grid_snapping.svg", _("Grid Snapping"), gui->get("menuSnapGrid")));
+								"grid_snapping" + darkThemeExt + ".svg", _("Grid Snapping"), gui->get("menuSnapGrid")));
```

Comment:
Do this on icon loading, and do it to a separate folder please, not much work.

You need to change only one place: GladeGui::loadIconPixbuf

Reply:
Hi @andreasb242 thanks for the reply, how should I do it in GladeGui::loadIconPixbuf? Should I pass a bool parameter in order to check a white icon? Then I should add a bool parameter in teh ToolButton  contructor too, is this what you're saying?

Reply:
Awesome, but I still have to manually select the icons which have the white theme. Not all the icons are in the "Dark theme" path, just the black ones, otherwise we would have to copy all the icons in the new path

Reply:
My fault, now I understand, if a file doesn't exist in the new path, search for it in the old one, my bad

Reply:
Yes, exactly. You load it only if it exists